datatree/io.py

```diff
@@ -68,16 +68,8 @@ def _open_datatree_netcdf(filename: str, **kwargs) -> DataTree:
         tree_root = DataTree.from_dict({"/": ds})
         for path in _iter_nc_groups(ncds):
             subgroup_ds = open_dataset(filename, group=path, **kwargs)
+            tree_root[path] = DataTree(name=path, data=subgroup_ds)
 
-            # TODO refactor to use __setitem__ once creation of new nodes by assigning Dataset works again
-            node_name = NodePath(path).name
-            new_node: DataTree = DataTree(name=node_name, data=subgroup_ds)
```

Comment:
Are you sure your new code behaves the same way as this old code? Because previously we were naming new nodes as `NodePath(path).name`, which would take the last part of a long unix-like path (e.g. `/a/b/c` -> `c`), whereas with your changes it looks like it would name the node `/a/b/c`?